src/deluge/gui/ui/sample_marker_editor.cpp

```diff
@@ -142,10 +145,28 @@ void SampleMarkerEditor::writeValue(uint32_t value, MarkerType markerTypeNow) {
 		getCurrentSampleHolder()->startPos = value;
 	}
 	else if (markerTypeNow == MarkerType::LOOP_START) {
-		getCurrentMultisampleRange()->sampleHolder.loopStartPos = value;
+		if (loopLocked) {
+			int intendedLoopEndPos = value + loopLength;
+			if (intendedLoopEndPos < getCurrentSampleHolder()->endPos) {
```

Comment:
What happens if the intended end pos is equal to the end of the sample? I think that's possible but unhandled

Reply:
I wasn't sure if the loop points could extend as far as sample star and end. I'll update so they can and test.

Reply:
Addressed in https://github.com/SynthstromAudible/DelugeFirmware/pull/293/commits/1b6dc94235e14b644c91bfe5d96965dab55fe28e